ext/zend_test/test.c

```diff
@@ -517,28 +517,110 @@ static bool has_opline(zend_execute_data *execute_data)
 	;
 }
 
+// When a custom memory manager is installed, ZendMM alters its behaviour. For
+// example `zend_mm_gc()` will not do anything anymore, or `zend_mm_shutdown()`
+// won't cleanup chunks and leak memory. This might not be a problem in tests,
+// but I fear folks might see and use this implementation as a boiler plate when
+// trying to use this hook
+int zend_test_prepare_zendmm_for_call(zend_mm_heap *heap)
+{
+	int flag = 0;
+	memcpy(&flag, heap, sizeof(int));
+	int new_flag = 0;
+	memcpy(heap, &new_flag, sizeof(int));
```

Comment:
Yeah, no.
It works, but we should not do it this way, especially not advertise that in php-src itself.

You probably should include a
```
#ifndef ZEND_ALLOC_HIDE_STRUCTS
struct _zend_mm_heap {
#if ZEND_MM_CUSTOM
    int use_custom_mm_heap;
#endif
}
#endif
```
to zend_alloc.h, with `#define ZEND_ALLOC_HIDE_STRUCTS 1` in zend_alloc.c

and have it public API.

If we rely on such implementation details, it can well just be public.

Reply:
Hmm, I get the point and the `zend_test_prepare_zendmm_for_call()` is super ugly 😉 
OTOH I would rather not make the `use_custom_mm_heap` flag public just for the sake of this test.

I am about to prepare a PR that would add a `gc` and `shutdown` custom handler (#13432) that might help solve this.